native/src/main/java/io/ballerina/stdlib/data/jsondata/FromString.java

```diff
@@ -0,0 +1,188 @@
+                case TypeTags.STRING_TAG:
+                    return string;
+                case TypeTags.BOOLEAN_TAG:
+                    return stringToBoolean(value);
+                case TypeTags.NULL_TAG:
+                    return stringToNull(value);
+                case TypeTags.UNION_TAG:
+                    return stringToUnion(string, (UnionType) expType);
+                case TypeTags.JSON_TAG:
+                    return stringToUnion(string, JSON_TYPE_WITH_BASIC_TYPES);
+                case TypeTags.TYPE_REFERENCED_TYPE_TAG:
+                    return fromStringWithType(string, ((ReferenceType) expType).getReferredType());
+                default:
+                    return returnError(value, expType.toString());
+            }
+        } catch (NumberFormatException e) {
+            return returnError(value, expType.toString());
+        }
+    }
+
+    private static Long stringToInt(String value) throws NumberFormatException {
+        return Long.parseLong(value);
+    }
+
+    private static Double stringToFloat(String value) throws NumberFormatException {
+        if (hasFloatOrDecimalLiteralSuffix(value)) {
+            throw new NumberFormatException();
+        }
+        return Double.parseDouble(value);
+    }
+
+    private static BDecimal stringToDecimal(String value) throws NumberFormatException {
+        return ValueCreator.createDecimalValue(value);
+    }
+
+    private static Object stringToBoolean(String value) throws NumberFormatException {
+        if ("true".equalsIgnoreCase(value) || "1".equalsIgnoreCase(value)) {
+            return true;
+        }
+
+        if ("false".equalsIgnoreCase(value) || "0".equalsIgnoreCase(value)) {
+            return false;
+        }
+        return returnError(value, "boolean");
+    }
+
+    private static Object stringToNull(String value) throws NumberFormatException {
+        if ("null".equalsIgnoreCase(value) || "()".equalsIgnoreCase(value)) {
+            return null;
+        }
+        return returnError(value, "()");
+    }
+
+    private static Object stringToUnion(BString string, UnionType expType) throws NumberFormatException {
+        List<Type> memberTypes = expType.getMemberTypes();
+        memberTypes.sort(Comparator.comparingInt(t -> TYPE_PRIORITY_ORDER.getOrDefault(
+                TypeUtils.getReferredType(t).getTag(), Integer.MAX_VALUE)));
+        boolean isStringExpType = false;
+        for (Type memberType : memberTypes) {
+            try {
+                Object result = fromStringWithType(string, memberType);
+                if (result instanceof BString) {
+                    isStringExpType = true;
+                    continue;
+                } else if (result instanceof BError) {
+                    continue;
+                }
+                return result;
+            } catch (Exception e) {
+                // Skip
+            }
+        }
+
+        if (isStringExpType) {
+            return string;
```

Comment:
isn't this should be return the `expType`. This method convert string to Union ryt?

Reply:
This method should be as below, I will update
```
private static Object stringToUnion(BString string, UnionType expType) throws NumberFormatException {
        List<Type> memberTypes = expType.getMemberTypes();
        memberTypes.sort(Comparator.comparingInt(t -> TYPE_PRIORITY_ORDER.getOrDefault(
                TypeUtils.getReferredType(t).getTag(), Integer.MAX_VALUE)));
        for (Type memberType : memberTypes) {
            try {
                Object result = fromStringWithType(string, memberType);
                if (result instanceof BError) {
                    continue;
                }
                return result;
            } catch (Exception e) {
                // Skip
            }
        }
        return returnError(string.getValue(), expType.toString());
    }
```